tests/00_00_04-change-federation.js

```diff
@@ -0,0 +1,246 @@
+
+    });
+
+    it('should commit the pending federation', async () => {
+
+        const pendingFederationHash = await bridge.methods.getPendingFederationHash().call();
+
+        expect(pendingFederationHash).to.be.equal(expectedPendingFederationHash, 'The pending federation hash should be the expected.').to.not.be.null;
+
+        const commitPendingFederationMethod = bridge.methods.commitFederation(pendingFederationHash);
+
+        // First commit pending federation vote
+        await rskUtils.sendTransaction(rskTxHelper, commitPendingFederationMethod, fedChangeAuthorizer1Address);
+
+        // Second commit pending federation vote
+        await rskUtils.sendTransaction(rskTxHelper, commitPendingFederationMethod, fedChangeAuthorizer2Address);
+
+        // Third and final commit pending federation vote
+        const commitFederationTransactionReceipt = await rskUtils.sendTransaction(rskTxHelper, commitPendingFederationMethod, fedChangeAuthorizer3Address);
+
+        const bridgeTransaction = await bridgeTxParser.getBridgeTransactionByTxHash(rskTxHelper.getClient(), commitFederationTransactionReceipt.transactionHash);
+
+        commitFederationEvent = bridgeTransaction.events.find(event => event.name === 'commit_federation');
+        expect(commitFederationEvent, 'The commit federation event should be emitted.').to.not.be.null;
+
+        // Assert the old federation address in the commit federation event is the initial one.
+        const oldFederationAddress = commitFederationEvent.arguments.oldFederationBtcAddress;
+        expect(oldFederationAddress).to.be.equal(initialFederationAddress, 'The old federation address in the commit_federation event should now be the same as the initial.');
+
+        // Assert the new federation address in the commit federation event is the expected one.
+        const newActiveFederationAddress = commitFederationEvent.arguments.newFederationBtcAddress;
+        expect(newActiveFederationAddress).to.be.equal(expectedNewFederationAddress, 'The new federation address in the commit_federation event should be the expected one.');
+
+        // Assert the new federation btc public keys in the commit federation event are the expected ones.
+        const expectedNewFederationBtcPublicKeys = newFederationPublicKeys.map(federator => federator[KEY_TYPE_BTC]);
+        const newFederationBtcPublicKeysInString = commitFederationEvent.arguments.newFederationBtcPublicKeys;
+        const actualNewFederationBtcPublicKeys = getBtcPublicKeysInArray(newFederationBtcPublicKeysInString);
+        expect(actualNewFederationBtcPublicKeys).to.be.deep.equal(expectedNewFederationBtcPublicKeys, 'The new federation btc public keys should be the expected ones.');
+
+    });
+
+    it('should activate federation', async () => {
+
+        const federationActivationBlockNumber = commitFederationEvent.arguments.activationHeight;
```

Comment:
I wouldn't get this value from the event. I think we should have a constant with the fed activation age, and assert that the event is emitted with the expected value

Reply:
Updated.

Reply:
Updated.